### PR TITLE
Deprecate JDate::toMySQL

### PR DESCRIPTION
**Pull request by chdemko**
In favor of JDate::toSQL

### CODE REVIEW

**Comment by ianmacl** [comment]
Do all databases expect the data to be in the same format?

**Comment by elinw** [comment]
In deprecation notices if possible please state what to use instead.

**Comment by chdemko** [comment]
@Ian: not necessarily: in my code, I call $dbo->getDateFormat() in the toSQL() method

**Comment by eddieajau** [comment]
I think the method name should be toSql, but other than at it looks fine.

**Comment by eddieajau** [comment]
Actually, can we also change all of the instances of toMysql to toSql in the platform code, so it's full deprecated in the API?

File: libraries/joomla/utilities/date.php [hunk 1]
```diff
@@ -523,14 +523,35 @@ public function toISO8601($local = false)
...
+	 *
+	 * @param   boolean    $local  True to return the date string in the local time zone, false to return it in GMT.
+	 * @param   JDatabase  $dbo    The database driver or null to use JFactory::getDbo()
+	 *
+	 * @return  string     The date string in SQL datetime format.
+	 *
+	 * @link http://dev.mysql.com/doc/refman/5.0/en/datetime.html
+	 * @since   11.4
+	 */
+	public function toSql($local = false, $dbo = null)
```
Comment (eddieajau) [inline libraries/joomla/utilities/date.php]: One minor thing:

```
public function toSql($local = false, JDatabase $dbo = null)
```

**Comment by eddieajau** [comment]
Awesome, I like it!

**Pull request merged**